Packs/SAP_BTP/Integrations/SAPBTP/SAPBTP.py

```diff
@@ -0,0 +1,708 @@
+            return handle
+        except IndexError:
+            demisto.debug("[Pagination] Failed to split handle from header. Assuming no next page.")
+            return None
+
+
+# endregion
+
+# region Command implementations
+# =================================
+# Command implementations
+# =================================
+
+
+def test_module(client: Client) -> str:
+    """Test API connectivity by fetching 1 minute of data."""
+    demisto.debug("[Test Module] Starting...")
+
+    try:
+        utc_now = datetime.now(timezone.utc)  # noqa: UP017
+        test_time = (utc_now - timedelta(minutes=Config.TEST_MODULE_LOOKBACK_MINUTES)).strftime(Config.DATE_FORMAT)
+
+        demisto.debug(f"[Test Module] Fetching from: {test_time}")
+        fetch_events_with_pagination(client, created_after=test_time, max_events=Config.TEST_MODULE_MAX_EVENTS)
+
+        demisto.debug("[Test Module] Success")
+        return "ok"
+
+    except Exception as error:
+        error_msg = str(error)
+        demisto.debug(f"[Test Module] Failed: {error_msg}")
+
+        if "401" in error_msg or "403" in error_msg:
+            return "Authorization Error: Verify Client ID, Secret, or Certificates."
+        raise
+
+
+def fetch_events_with_pagination(
+    client: Client, created_after: str, max_events: int, created_before: str | None = None
+) -> list[dict[str, Any]]:
+    """Fetch, Sort (Oldest First), and Slice events.
+
+    Implements "Fetch-Sort-Slice" strategy:
+    - Fetch pages until we have at least 'max_events' raw items.
+    - Sort raw list by time (Oldest -> Newest).
+    - Slice to return exactly 'max_events' (the oldest ones).
+    """
+    events: list[dict[str, Any]] = []
+    pagination_handle: str | None = None
+    page_count = 0
+
+    demisto.debug(f"[Pagination Loop] Start. Goal: {max_events}. Time: {created_after} -> {created_before or 'Now'}")
+
+    # Fetch Loop
+    while len(events) < max_events:
+        page_count += 1
+
+        page_events, pagination_handle = client.get_audit_log_events(
+            created_after=created_after, created_before=created_before, pagination_handle=pagination_handle
+        )
+
+        if not page_events:
+            demisto.debug(f"[Pagination Loop] Page {page_count}: Empty. Stopping.")
+            break
+
+        events.extend(page_events)
+        demisto.debug(f"[Pagination Loop] Page {page_count}: +{len(page_events)} events. Total accumulated: {len(events)}")
+
+        if not pagination_handle:
+            demisto.debug("[Pagination Loop] No next page handle. Stopping.")
+            break
+
+        # Stop fetching if we have enough
+        if len(events) >= max_events:
+            demisto.debug(f"[Pagination Loop] Threshold reached ({len(events)} >= {max_events}). Stopping fetch.")
+            break
+
+    if not events:
+        demisto.debug("[Pagination Result] No events found.")
+        return []
+
+    # Sort events by time (Oldest First)
```

Comment:
cant we get the results already sorted as part of the request? sorting is expensive.

Reply:
Unfortunately, no